src/App.js

```diff
@@ -1,26 +1,46 @@
 import {nanoid} from 'nanoid';
-import {useState, useRef} from 'react';
+import {useState, useEffect, useRef} from 'react';
 import styled from 'styled-components';
 
 import {mazeArray} from './arrays/mazeArray.js';
-import MagicForm from './components/MagicForm/MagicForm';
+import MagicForm from './components/MagicForm/MagicForm.js';
+import MagicImage from './components/MagicImage/MagicImage.js';
 import MagicList from './components/MagicList/MagicList.js';
-import {CodeSpan} from './components/Stylesheet/StyledSpans.js';
 import executeSpell from './functionsfolder/executeSpell.js';
 import findCommandMessage from './functionsfolder/findCommandMessage.js';
+import mazeEnd from './images/mazeRooms/mazeEnd.png';
+import mazeWay from './images/mazeRooms/mazeWay.png';
+import {BackgroundImg} from './stylesheet/StyledImages.js';
+import {CodeSpan} from './stylesheet/StyledSpans.js';
 
 export default function App() {
   const [currentArrayPosition, setCurrentArrayPosition] = useState(mazeArray[0]); //this holds a {object}
 
   function changePosition(newPositionAsString) {
     const newPositionAsObject = mazeArray.find(mazeRoom => mazeRoom.path === newPositionAsString);
     setCurrentArrayPosition(newPositionAsObject);
+    setIsRoomEnlighten(false);
   }
 
   const [savedOrders, setSavedOrders] = useState([]);
 
   const ref = useRef();
 
+  const [isRoomEnlighten, setIsRoomEnlighten] = useState(false);
+  function changeEnlighten() {
+    setIsRoomEnlighten(true);
+  }
+  useEffect(() => {}, [isRoomEnlighten]); //used to rerender idependly from useEffect below
+
+  const [currentBackgroundImage, setCurrentBackgroundImage] = useState();
+  useEffect(() => {
+    if (currentArrayPosition.type === 'start' || currentArrayPosition.type === 'end') {
+      setCurrentBackgroundImage(<BackgroundImg src={mazeEnd} alt="the end of the way" />);
+    } else {
+      setCurrentBackgroundImage(<BackgroundImg src={mazeWay} alt="a lonely and spooky corridor" />);
+    }
+  }, [currentArrayPosition]);
```

Comment:
For better readability, the states of a component are often right at the top of the component, followed by the `useEffect` hooks and then by the different functions. I personally prefer it that way as you can directly see how many states there are and what happens when (useEffect).
```suggestion
  const [currentArrayPosition, setCurrentArrayPosition] = useState(mazeArray[0]); //this holds a {object}
  const [currentBackgroundImage, setCurrentBackgroundImage] = useState();
  const [savedOrders, setSavedOrders] = useState([]);
  const [isRoomEnlighten, setIsRoomEnlighten] = useState(false);
  
   const ref = useRef();
   
   useEffect(() => {}, [isRoomEnlighten]); //used to rerender idependly from useEffect below

  useEffect(() => {
    if (currentArrayPosition.type === 'start' || currentArrayPosition.type === 'end') {
      setCurrentBackgroundImage(<BackgroundImg src={mazeEnd} alt="the end of the way" />);
    } else {
      setCurrentBackgroundImage(<BackgroundImg src={mazeWay} alt="a lonely and spooky corridor" />);
    }
  }, [currentArrayPosition]);

  function changePosition(newPositionAsString) {
    const newPositionAsObject = mazeArray.find(mazeRoom => mazeRoom.path === newPositionAsString);
    setCurrentArrayPosition(newPositionAsObject);
    setIsRoomEnlighten(false);
  }



  function changeEnlighten() {
    setIsRoomEnlighten(true);
  }
```